stumpy/ostinato.py

```diff
@@ -0,0 +1,187 @@
+
+    ts_ind : int
+        Index of time series which contains the consensus motif
+
+    ss_ind : int
+        Start index of consensus motif within the time series ts_ind
+        that contains it
+
+    Notes
+    -----
+    <https://www.cs.ucr.edu/~eamonn/consensus_Motif_ICDM_Long_version.pdf>
+
+    See Table 2
+
+    The proposed ostinato algorithm only finds the best radius but not the
+    most central motif. If you need the most central motif run
+    stumpy.ostinato._get_central_motif with this function's return values.
+    """
+    # Preprocess means and stddevs and handle np.nan/np.inf
+    Ts = [None] * len(tss)
+    M_Ts = [None] * len(tss)
+    Σ_Ts = [None] * len(tss)
+    for i, T in enumerate(tss):
+        Ts[i], M_Ts[i], Σ_Ts[i] = core.preprocess(T, m)
+
+    bsf_rad, ts_ind, ss_ind = np.inf, 0, 0
+    k = len(Ts)
+    for j in range(k):
+        if j < (k - 1):
+            h = j + 1
+        else:
+            h = 0
+
+        mp = stump(Ts[j], m, Ts[h], ignore_trivial=False)
+        si = np.argsort(mp[:, 0])
+        for q in si:
+            rad = mp[q, 0]
+            if rad >= bsf_rad:
+                break
+            for i in range(k):
+                if ~np.isin(i, [j, h]):
+                    QT = core.sliding_dot_product(Ts[j][q : q + m], Ts[i])
+                    rad = np.max(
+                        (
+                            rad,
+                            np.min(
+                                core._mass(
+                                    Ts[j][q : q + m],
+                                    Ts[i],
+                                    QT,
+                                    M_Ts[j][q],
+                                    Σ_Ts[j][q],
+                                    M_Ts[i],
+                                    Σ_Ts[i],
+                                )
+                            ),
+                        )
+                    )
+                    if rad >= bsf_rad:
+                        break
+            if rad < bsf_rad:
+                bsf_rad, ts_ind, ss_ind = rad, j, q
+
+    return bsf_rad, ts_ind, ss_ind
+
+
+def _get_central_motif(ts, rad, tsi, ssi, m):
+    """
+    Compare subsequences with the same radius and return the most central motif
+
+    Parameters
+    ----------
+    ts : list
+        List of time series for which to find the most central motif
+
+    rad : float
+        Best radius found by a consensus search algorithm
+
+    tsi : int
+        Index of time series in which `rad` was found first
+
+    ssi : int
+        Start index of subsequence in `tsi` that has radius `rad`
+
+    m : int
+        Window size
+
+    Returns
+    -------
+    rad : float
+        Radius of the most central consensus motif
+
+    tsi : int
+        Index of time series which contains the most central consensus motif
+
+    ssi : int
+        Start index of most central consensus motif within the time series `tsi`
+        that contains it
```

Comment:
Since this is not covered in the original published work, would you mind providing a detailed note on what is happening in this function and what it is trying to achieve? This will be helpful for others in the future.

Reply:
I added a detailed note. Please let me know what you think. Maybe there's a way to make it more concise.

Reply:
Looks good. Thank you